unsigned_commands.go

```diff
@@ -3,70 +3,42 @@ package main
 import (
 	"strings"
 	"path/filepath"
+	"runtime"
 	"fmt"
 	"os"
 )
 
-func getUploadCommand(command string) (string, bool) {
+const (
+	posixSpecialChars = "!\"#$&'()*,;<=>?[]\\^`{}|~"
+	batchSpecialChars = "^&;,=%"
+)
+
+func isUploadCommand(command string) bool {
 	// buildkite-signed-pipeline upload
 	rawUploadCommand := fmt.Sprintf("%s upload", filepath.Base(os.Args[0]))
 	if strings.HasPrefix(command, rawUploadCommand) {
-		return rawUploadCommand, true
+		return true
 	}
 
-	vanillaUploadCommand := "buildkite-agent pipeline upload"
-	if strings.HasPrefix(command, vanillaUploadCommand) {
-		return vanillaUploadCommand, true
+	// vanilla upload command
+	if strings.HasPrefix(command, "buildkite-agent pipeline upload") {
+		return true
 	}
 
-	return "", false
+	return false
 }
 
-func IsUnsignedCommandOk(command string) (bool, error) {
-	uploadCommand, ok := getUploadCommand(command)
-	if !ok {
-		return false, nil
-	}
-
-	// straight upload
-	if uploadCommand == command {
-		return true, nil
+func hasSpecialShellChars(str string) bool {
+	if runtime.GOOS == `windows` {
```

Comment:
I'd just check both, otherwise you might have the pipeline upload on windows, but later steps on linux.

Reply:
Isn't that OK? as the "current" agent would be the one executing the unsigned upload command?

Reply:
Yeah, fair point.